### ISSUE TITLE
Tasks should be assigned depth first, not breadth first

### ISSUE THREAD

**Issue by krzyk** [zerocracy/farm#1211]
To speed up delivery of features/bugfixes it would be best to have a semi depth-first assignment of tasks that are generated from puzzles.

The above is most probably hard/impossible to do right now, so the easiest solution right now would be to first assign tasks that were generated by 0pdd and then the rest that are in the project.

(Ideal solution would be: E.g. When I have a task #10, and create a puzzles there which will generate task #20 and #30. Those tasks #20 and #30 should be assigned before any other tasks in the project. And if those tasks generate new puzzles, those new puzzles should be also assigned first. Basically build a graph of tasks and their children/grandchildrent/etc. and assign the youngest descendants first)

**Comment by g4s8** [comment]
@krzyk I think we can create new election `Rank` to move tasks with `pdd` labels up, similar to bug label.

**Comment by 0crat** [comment]
Job [#1211](https://github.com/zerocracy/farm/issues/1211) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by krzyk** [comment]
@g4s8 Hmm, I think all pdd tasks are also bugs (I think that is result of .0pdd.yml configuration). And it doesn't look like bugs are assigned first.

**Comment by g4s8** [comment]
@krzyk we have `RnkGithubBug` for it. Now all review tasks goes first, then boosted tasks, then tasks with `bug` label

**Comment by 0crat** [comment]
The job [#1211](https://github.com/zerocracy/farm/issues/1211) assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2660644288); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by krzyk** [comment]
@krzyk done in #1722

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1211](https://github.com/zerocracy/farm/issues/1211) is now out of scope

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
